README.md

```diff
@@ -0,0 +1,89 @@
+# it-multistream-select
+
+## "Dialer"
+
+```js
+const pipe = require('it-pipe')
+const MSS = require('it-multistream-select')
+const Mplex = require('it-mplex') // https://github.com/alanshaw/it-mplex
+
+const muxer = new Mplex()
+const muxedStream = muxer.newStream()
+
+// MSS.select(stream, protocol(s))
+// Select from one of the passed protocols (in priority order)
+// Returns selected stream and protocol
+const { stream: dhtStream, protocol } = await MSS.select(muxedStream, [
+  // This might just be different versions of DHT, but could be different impls
+  '/ipfs-dht/2.0.0', // Most of the time this will probably just be one item.
+  '/ipfs-dht/1.0.0'
+])
+
+// Typically this stream will be passed back to the caller of libp2p.dialProtocol
```

Comment:
Ideally with the change to async I'd actually like to eventually migrate to a `const dhtStream = libp2p.newStream(peerInfo, [...protocols])` call. This would allow us to do more discrete `connect` vs `protocol stream` calls. This shouldn't change the behavior here though.